building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,541 @@
+
+With the manifest in place, create a Dockerfile to house the build instruction for the application. Dockerfile contains build instruction for images to be deployed on Okteto.
+
+```sh
+$ touch Dockerfile
+```
+
+Add the following:
+
+```Dockerfile
+FROM python:3.8
+
+ADD requirements.txt /requirements.txt
+
+ADD main.py /main.py
+
+ADD okteto-pipeline.yaml /okteto-pipeline.yaml
+
+ADD okteto-stack.yaml /okteto-stack.yaml
+
+ADD okteto.yml /okteto.yml
+
+RUN pip install -r requirements.txt
+
+EXPOSE 8080
+
+COPY ./app app
+
+CMD ["python3", "main.py"]
+```
+
+Next, create an `okteto-pipeline.yaml` file. The Okteto pipeline file gives you more control over how you want to deploy and build your application. Add the following in the pipeline file:
+
+```yaml
+deploy:
+  - okteto stack deploy --build
+dev:
+  - okteto.yml
+```
+
+In the pipeline file above, you defined the instruction for deployment and development. When Okteto deploys your application, it references the instruction listed in the stack manifest, and when it's developing the application, it references the Okteto manifest for instructions.
+
+Before proceeding to deployment on Okteto, download and set up your Kubernetes credentials on your local machine:
+
+![Okteto Dashboard](https://res.cloudinary.com/adeshina/image/upload/v1608635743/x9tmv4styid2hffxilxx.png)
+
+Let's deploy your application to Okteto. Start by running the command: 
+
+```
+okteto namespace
+```
+
+The command above setups a namespace on the Okteto cloud. Next, run the command:
+
+```
+okteto up
+```
+
+The command above setups a new deployment. You should have a similar response in your console:
+
+```
+Deployment fastapi-crud doesn't exist in namespace youngestdev. Do you want to create a new one? [y/n]: y
+ ✓  Development container activated
+ ✓  Connected to your development container
+ ✓  Files synchronized
+    Namespace: youngestdev
+    Name:      fastapi-crud
+
+Welcome to your development container. Happy coding!
+
+```
+
+Log on to your Okteto dashboard; you should see the newly deployed application alongside a volume created similar to this:
+
+![Deployed Application](https://res.cloudinary.com/adeshina/image/upload/v1608636581/jq7fjtldhz1tqqzuh29h.png)
+
+In your console, you are automatically logged into your cloud server. So, return to the terminal and create a virtual environment:
+
+```sh
+youngestdev:fastapi-crud app> python3 -m venv venv
+```
+
+If the above command throws an error, install the venv module:
+
+```sh
+youngestdev:fastapi-crud app> apt-get install python3-venv -y
+```
+
+Rerun the first command afterward. Activate the virtual environment and install the dependencies:
+
+```sh
+youngestdev:fastapi-crud app> source venv/bin/activate
+(venv) youngestdev:fastapi-crud app> pip install -r requirements.txt
+```
+
+Next, start the application:
+
+```
+(venv) youngestdev:fastapi-crud app> python3 main.py
+```
```

Comment:
Instead of this, we can just use `okteto stack deploy --build` to achieve the same with less commands.